task/upload.go

```diff
@@ -216,12 +247,39 @@ func TaskTranscodeFile(tctx *TaskContext) (*TaskHandlerOutput, error) {
 			}
 			return &TaskHandlerOutput{TaskOutput: &data.TaskOutput{TranscodeFile: &tfo}}, nil
 		},
-		profiles:                 params.Profiles,
 		catalystPipelineStrategy: pipeline.Strategy(params.CatalystPipelineStrategy),
 		targetSegmentSizeSecs:    params.TargetSegmentSizeSecs,
 	})
 }
 
+func TaskClip(tctx *TaskContext) (*TaskHandlerOutput, error) {
+	params := *tctx.Task.Params.Clip
+	return handleUploadVOD(handleUploadVODParams{
+		tctx:  tctx,
+		inUrl: params.URL,
+		getOutputLocations: func() ([]clients.OutputLocation, error) {
+			_, outputLocations, err := clipTaskOutputLocations(tctx)
+			return outputLocations, err
+		},
+		finalize: func(callback *clients.CatalystCallback) (*TaskHandlerOutput, error) {
+			tctx.Progress.Set(0.9)
+			taskOutput, err := processCatalystCallback(tctx, callback)
+			if err != nil {
+				return nil, fmt.Errorf("error processing catalyst callback: %w", err)
+			}
+			return &TaskHandlerOutput{
+				TaskOutput: &data.TaskOutput{Upload: taskOutput},
```

Comment:
This should be `Clip` or API will break processing this result.
```suggestion
				TaskOutput: &data.TaskOutput{Clip: taskOutput},
```